Riot/Modules/Onboarding/OnboardingCoordinator.swift

```diff
@@ -193,39 +197,46 @@ final class OnboardingCoordinator: NSObject, OnboardingCoordinatorProtocol {
         
         guard BuildSettings.onboardingEnableNewAuthenticationFlow else {
             showLegacyAuthenticationScreen()
+            coordinator.stop()
             return
         }
         
-        if result == .customServer {
-            beginAuthentication(with: .selectServerForRegistration)
-        } else {
-            beginAuthentication(with: .registration)
+        Task {
+            if result == .customServer {
+                await beginAuthentication(with: .selectServerForRegistration)
+            } else {
+                await beginAuthentication(with: .registration)
+            }
+            
+            coordinator.stop()
         }
     }
     
     // MARK: - Authentication
     
     /// Show the authentication flow, starting at the specified initial screen.
-    private func beginAuthentication(with initialScreen: AuthenticationCoordinator.EntryPoint) {
+    private func beginAuthentication(with initialScreen: AuthenticationCoordinator.EntryPoint) async {
         MXLog.debug("[OnboardingCoordinator] beginAuthentication")
         
         let parameters = AuthenticationCoordinatorParameters(navigationRouter: navigationRouter,
                                                              initialScreen: initialScreen,
                                                              canPresentAdditionalScreens: false)
         let coordinator = AuthenticationCoordinator(parameters: parameters)
-        coordinator.completion = { [weak self, weak coordinator] result in
+        coordinator.callback = { [weak self, weak coordinator] result in
             guard let self = self, let coordinator = coordinator else { return }
             
             switch result {
             case .didLogin(let session, let authenticationFlow, let authenticationType):
                 self.authenticationCoordinator(coordinator, didLoginWith: session, and: authenticationFlow, using: authenticationType)
             case .didComplete:
                 self.authenticationCoordinatorDidComplete(coordinator)
+            case .cancel(let flow):
+                self.cancelAuthentication(flow: flow)
             }
         }
         
         add(childCoordinator: coordinator)
-        coordinator.start()
+        await coordinator.startAsync()
```

Comment:
I suppose this is where we would call the sync variant, otherwise we are creating an inconsistency with the default `Coordinator` protocol which specifies a generic sync `start` method. Is there any downside of doing that?